beacon-chain/core/state/transition_stategen.go

```diff
@@ -0,0 +1,85 @@
+package state
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	stateTrie "github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"go.opencensus.io/trace"
+)
+
+// ExecuteStateTransitionStateGen applies state transition on input historical state and block for state gen usages.
+// There's no signature verification involved given state gen only works with stored block and state in DB.
+// If the objects are already in stored in DB, one can omit redundant signature checks and ssz hashing calculations.
+// WARNING: This method should not be used for new block.
```

Comment:
Can this be private? This warning makes me a bit nervous since it is not enforced

Reply:
Great point. I have moved it to `stategen` package and made it private